CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+cmake_minimum_required(VERSION 3.10)
+include(ExternalProject)
+
+if (DEFINED CIMEROOT)
+  message("Using CIME in ${CIMEROOT} with compiler ${COMPILER}")
+  include(${CASEROOT}/Macros.cmake)
+  if (${PIO_VERSION} LESS 2)
+    message( FATAL_ERROR "Version 2 of the PIO library required")
+  endif()
+  if (${MPILIB} STREQUAL "mpi-serial")
+    set(CMAKE_C_COMPILER ${SCC})
+    set(CMAKE_Fortran_COMPILER ${SFC})
+    set(CMAKE_CXX_COMPILER ${SCXX})
+  else()
+    set(CMAKE_C_COMPILER ${MPICC})
+    set(CMAKE_Fortran_COMPILER ${MPIFC})
+    set(CMAKE_CXX_COMPILER ${MPICXX})
+  endif()
+  set(CMAKE_Fortran_FLAGS "${FFLAGS} -I${LIBROOT}/include -I${LIBROOT}/finclude -I${LIBROOT}/nuopc/esmf/${NINST_VALUE}/include")
+else()
+  set(BLD_STANDALONE TRUE)
+endif()
+
+project(CMEPS LANGUAGES Fortran VERSION 0.1)
+
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
+
+message("CMAKE_MODULE_PATH is ${CMAKE_MODULE_PATH}")
+
+find_package(ESMF REQUIRED)
+if (DEFINED PIO)
+  set(PIO_PATH ${PIO})
+else()
+  set(PIO_PATH $ENV{PIO})
+endif()
+find_package(PIO REQUIRED COMPONENT C Fortran PATH ${PIO_PATH})
+
+if (NOT DEFINED MPILIB OR NOT ${MPILIB} STREQUAL "mpi-serial")
+  find_package(MPI REQUIRED)
+endif()
+
+if(BLD_STANDALONE)
+  add_subdirectory(nems/util)
+  list(APPEND EXTRA_LIBS cmeps_share)
+  list(APPEND EXTRA_INCLUDES "${CMAKE_BINARY_DIR}/nems/util")
+endif()
+
+add_subdirectory(mediator)
```

Comment:
So the cmake build is available in CMEPS master now? Is it same as our cmake CMakeLists.txt

Reply:
I think escomp has always had a cmake build. I see that we probably have a conflict with how we've implemented it. I can manually remove their cmake files (update only mediator source code) but I'm not sure how to manage this in the future. If we PR back to escomp, then we'll constantly have to manage this?

Reply:
That's what we want to avoid. Maybe we need to have a meeting with CESM to discuss if we can have one cmake build that works for all the apps

Reply:
I think that makes sense. But I think we should merge this update (manually managing the conflicts) because it does contain a bugfix which we should get committed. Then we can still pursue a common path forward w/ cesm for the cmake build.